### ISSUE TITLE
Figure out correct way to export components in ES6 modules for Babel to work with ES5 code

### ISSUE THREAD

**Issue by jchansen** [lore/lore#71]
When babel was upgraded from v5 to v6, they changed the transpiling result of the `export default` function.  You used to be able to do this:

```js
// a.js
exports default 'some code';
```

```js
// b.js
var someCode = require(`./a`);
```

And everything was fine.  But (from what I understand) people were using the import statements incorrectly, and the transpiling behavior was allowing code to work that was actually in conflict with the ES6 spec. So in v6, the code above no longer works, and now you have to add `.defaults` to the require statement like this:

```js
// a.js
exports default 'some code';
```

```js
// b.js
var someCode = require(`./a`).defaults;
```

Somehow/somewhere there's a proper way to export and import in babel v6 that conforms to ES6 but also works with `require` statements and ES5 code, so you can mix and match ES5 and ES6 (or transpiled and not transpiled) code as you need to.

Need to figure that out and update the framework's ES6 code appropriate, and change any ES5 code that has incompatible behaviors.  Adding `.defaults` to require statements is not an acceptable solution.

While there is this [a babel-plugin-add-module-exports plugin for babel](babel-plugin-add-module-exports) that would remove the issue in v6, I believe it's intended to supply the old behavior only as a interim solution to buy more time (seeing as the behavior is not in spec in the first place).

**Comment by jchansen** [comment]
#107 makes it possible to create an ES6 project using `lore new app-name --es6` and sets a flag in the `.lorerc` file of the project that notifies all generators to create an ES6 version of whatever they produce as well.

The pattern it uses to do that will also easily support creating an ES.Next experience once I confirm the lore.connect decorator works using the @lore.connect syntax, and then we can have the generators use static types and decorators and whatever other else is useful but still in a draft stage.

As for the original subject of this issue, I was able to solve most of the issues by ES5/ES6 interoperability issues by using the [babel-plugin-add-module-exports](https://www.npmjs.com/package/babel-plugin-add-module-exports), but not all of them. There was one place in `lore` I needed to add a line [to detect and use .default if it existed](https://github.com/lore/lore/blob/9da3014e7d0514d0d3d1b86aaa9f7a54a7c3d0d1/packages/lore/src/app/index.js#L101). 

I had to do the same in [webpack-requiredir](https://github.com/lore/webpack-requiredir/blob/a53245d0127bacc9982ba3c36ee6542dc5a01345/src/index.js#L23-L25) to get the config to load correctly. I don't understand why, but I gave up trying to figure it out after a while. I'll remove the hack in the future when it's no longer needed.

Also, while Node v6.2 has 96% compatibility with ES6, it looks like we're still waiting for v8 to support import/export syntax, which is waiting on the [module loader](https://github.com/whatwg/loader). No ETA that I can find, it's just "in progress".

Once v8 (and Node) support the import/export for module loading, I'll be open to switching the default to ES6. As it is right now, using ES6 modules breaks CLI tools like `gulp` that use vanilla node, and webpack only works if you kick it off with `babel-node`, which appears to have a negative impact on the development experience (loading times) and prevents setting breakpoints during testing (at least in IDEs, not sure about elsewhere).

**Comment by jchansen** [comment]
So given the changes above, I think the last thing that remains before I can close out this issue is upgrading React-Router to v2 and adding support for it in the ES6 generator templates for components.

**Comment by jchansen** [comment]
Closing this as #109 updates `react-router` to v2 and integrates routing into the ES6 templates.
